docs/features/incremental-generators.cookbook.md

```diff
@@ -636,7 +636,248 @@ TODO: https://github.com/dotnet/roslyn/issues/72149
+            {
+                if (!attribute.Name.ToString().Equals(AutoImplementAttributeName) || attribute.ArgumentList is null)
+                    continue;
+
+                return attribute.ArgumentList.Arguments
+                    .Select(x => GetInterfaceName(x.Expression.ToString()))
+                    .ToArray();
+            }
+        }
+
+        return [];
+    }
+    private static string GetInterfaceName(string expression)
+    {
+        // Empty String
+        if (string.IsNullOrWhiteSpace(expression))
+            return expression;
+
+        // Interface Name got by nameof()
+        if (expression.StartsWith(NameOfString))
+            return expression.Substring(NameOfString.Length, expression.Length - NameOfString.Length - 1);
+
+        //Trimming ""
+        string ret = expression.Trim('"');
+        return ret;
+    }
+    private static INamedTypeSymbol? GetInterfaceSymbol(Compilation compilation, string[] nameSpaces, string implementedInterfaceName)
+    {
+        // Try Get without NameSpace if interface was already Fully Qualified
+        INamedTypeSymbol? interfaceSymbol = compilation.GetTypeByMetadataName(implementedInterfaceName);
+        if (interfaceSymbol is not null)
+            return interfaceSymbol;
+
+        foreach (string nameSpace in nameSpaces)
+        {
+            interfaceSymbol = compilation.GetTypeByMetadataName($"{nameSpace}.{implementedInterfaceName}");
+            if (interfaceSymbol is not null)
+                return interfaceSymbol;
+        }
+
+        return null;
+    }
+
+    private record Model(string ClassName, string ClassNameSpace, InterfaceModel[] Interfaces);
+    private record InterfaceModel(string Name, string NameSpace, string[] Usings, InterfacePropertyModel[] Properties);
+    private record InterfacePropertyModel(string Type, string Name, bool HasSetter);
+
+    private void Execute(SourceProductionContext context, ImmutableArray<Model> models)
+    {
+        foreach (Model model in models)
+        {
+            foreach (InterfaceModel interfaceModel in model.Interfaces)
+            {
+                string interfaceUsings = string.Join(string.Empty, interfaceModel.Usings);
+
+                StringBuilder sourceBuilder = new($$"""
+                    {{interfaceUsings}}
+                    using {{interfaceModel.NameSpace}};
+
+                    namespace {{model.ClassNameSpace}};
+
+                    partial class {{model.ClassName}} : {{interfaceModel.Name}}
+                    {
+
+                    """);
+                foreach (InterfacePropertyModel property in interfaceModel.Properties)
+                {
+                    //Check if property has a setter
+                    string setter = property.HasSetter
+                        ? "set; "
+                        : string.Empty;
+
+                    sourceBuilder.AppendLine($$"""
+                            public {{property.Type}} {{property.Name}} { get; {{setter}}}
```

Comment:
Since this is only auto-implementing properties as `get; set;`, it would probably be good to narrow the name of `AutoImplementAttribute` to convey that up front for people reading the sample.